policy-report/kube-bench-adapter/pkg/kubebench/kubebench.go

```diff
@@ -0,0 +1,177 @@
+package kubebench
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	kubebench "github.com/aquasecurity/kube-bench/check"
+	batchv1 "k8s.io/api/batch/v1"
+	apiv1 "k8s.io/api/core/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	yaml "k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog"
+)
+
+func getClientSet(kubeconfigPath string) (*kubernetes.Clientset, error) {
+	var kubeconfig *rest.Config
+
+	cfg, err := rest.InClusterConfig()
+	if err != nil {
+		cfg, err = clientcmd.BuildConfigFromFlags("", kubeconfigPath)
+		if err != nil {
+			klog.Fatalf("Error building kubeconfig: %s", err.Error())
+			return nil, err
+		}
+	}
+	kubeconfig = cfg
+
+	clientset, err := kubernetes.NewForConfig(kubeconfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return clientset, nil
+
+}
+func RunJob(kubeconfig string, kubebenchYAML, kubebenchImg, kubebenchVersion, kubebenchBenchmark, kubebenchTargets string, timeout time.Duration) (*kubebench.OverallControls, error) {
+
+	clientset, err := getClientSet(kubeconfig)
+	if err != nil {
+		return nil, err
+	}
+	var jobName string
+	jobName, err = deployJob(context.TODO(), clientset, kubebenchYAML, kubebenchImg, kubebenchVersion, kubebenchBenchmark, kubebenchTargets)
```

Comment:
why do we create a `context.TODO` vs a `context.Background`? should we create a single context and pass to all calls?

Reply:
Hi @JimBugwadia , thanks for your suggestion, I acknowledge that after reading the documentation, it will be better to use `context.Background` and I am fixing it, however, I didn't understand what do we mean by creating a single context? Do we mean creating a global variable of context type?